src/resolver/descriptor.cc

```diff
@@ -457,6 +457,69 @@ to_json( nlohmann::json & jto, const ManifestDescriptorRaw & descriptor )
     }
 }
 
+/* -------------------------------------------------------------------------- */
+
+ManifestDescriptorRaw::ManifestDescriptorRaw(
+  const std::string_view & descriptor )
+{
+  size_t cursor = 0;
+  // Grab the input if it exists
+  if ( auto inputSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::inputSigil );
+       inputSepIdx != std::string_view::npos )
+    {
+      this->packageRepository
+        = std::string( descriptor.substr( cursor, inputSepIdx - cursor ) );
+      cursor = inputSepIdx + 1;
+    }
+  // Grab the attribute path or package name
+  size_t attrsEndIdx;
+  bool   hasVersion = false;
+  if ( auto versionSepIdx
+       = descriptor.find_first_of( ManifestDescriptorRaw::versionSigil );
+       versionSepIdx != std::string_view::npos )
+    {
+      attrsEndIdx = versionSepIdx;
+      hasVersion  = true;
+    }
+  else { attrsEndIdx = descriptor.size(); }
+  auto maybeAttrs
+    = std::string( descriptor.substr( cursor, attrsEndIdx - cursor ) );
+  auto numAttrSeparators
+    = std::count_if( maybeAttrs.begin(),
+                     maybeAttrs.end(),
+                     []( auto attrChar ) { return attrChar == '.'; } );
+  switch ( numAttrSeparators )
+    {
+      case 0:
+        // We were given a `pname`
+        this->name = maybeAttrs;
+        break;
+      case 1:
```

Comment:
suggestion blocking: I think you need to check prefixes, not just count `.`. E.g. I think this would classify `linuxKernel.packages.linux_4_19` as an absolute path even though it's relative.

Reply:
Fixed